### PR TITLE
add replica estimator interface

### PR DESCRIPTION
**Pull request by dixudx**
Signed-off-by: Di Xu <stephenhsu90@gmail.com>

#### What type of PR is this?
kind/apis
kind/feature

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Comment by yinsenyan** [comment]
replicas怎么下去？

**Comment by Garrybest** [comment]
There is no need to pass replicas into estimator. The estimator only needs a pod template and returns the max replicas.

File: pkg/apis/scheduler/types.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty" protobuf:"bytes,2,opt,name=tolerations"`
+
+	// Affinity specifies the scheduling constraints of a new replica.
+	// +optional
+	Affinity corev1.Affinity `json:"affinity,omitempty" protobuf:"bytes,3,opt,name=affinity"`
+
+	// Resources describes the compute resource requirements.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,4,opt,name=resources"`
```
Comment (Garrybest) [inline pkg/apis/scheduler/types.go:41]: Note that we only concern about request resources.

File: pkg/apis/scheduler/types.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+*/
+
+package scheduler
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ReplicaRequirements describes the scheduling requirements for a new replica.
+type ReplicaRequirements struct {
```
Comment (Garrybest) [inline pkg/apis/scheduler/types.go:24]: Could reuse APIs in #283.

Comment (dixudx) [reply pkg/apis/scheduler/types.go:24]: Sure.

**Comment by dixudx** [comment]
Will rebase and modify this PR after v0.9.0 is shipped.

**Pull request merged**